apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.UnitTests/PriorityListTests.cs

```diff
@@ -0,0 +1,235 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Google.Cloud.Spanner.V1;
+using Xunit;
+
+namespace Google.Cloud.Spanner.Data.Tests
+{
+    public class PriorityListTests
+    {
+        [Fact]
+        public void CanAddMultipleOfSamePriority()
+        {
+            PriorityList<TestItem> priorityList = new PriorityList<TestItem>();
+            TestItem item1, item2;
+            priorityList.Add(item1 = new TestItem(5));
+            priorityList.Add(item2 = new TestItem(5));
+            Assert.Equal(2, priorityList.Count);
+            AssertExist(priorityList, item1, item2);
+        }
+
+        [Fact]
+        public void DeleteRemovesCorrectItem()
+        {
+            PriorityList<TestItem> priorityList = new PriorityList<TestItem>();
+            TestItem toRemove;
+            priorityList.Add(toRemove = new TestItem(5));
+            priorityList.Add(new TestItem(5));
```

Comment:
I'd add another item with the same priority first, like this:
```csharp
priorityList.Add(new TestItem(5));
priorityList.Add(toRemove = new TestItem(5));
priorityList.Add(new TestItem(5));
```
That way you'll catch implementations in the future which erroneously pick the first or the last item with the specified priority.